### ISSUE TITLE
Linux: selected directory ending with \n

### ISSUE THREAD

**Issue by casperstorm** [balthild/native-dialog-rs#8]
Hi @balthild,
This is a great crate, so thanks. Keep up the good work.
I have a little project where a user has reported a problem on Linux:

> When I select dir from button "Select directory" the selected path end with newline and the addons directory cannot be found.
Had to do ugly workaround to get it working:

```rust
async fn open_directory() -> Option<PathBuf> {
    // Should we use task::spawn_blocking here?
    let dialog = OpenSingleDir { dir: None };
    dialog.show().unwrap().and_then(|f| Some(PathBuf::from(&f.as_path().display().to_string().replace("\n", ""))))
}
```

So question, is this a behaviour you expect or is it a bug? You can find the original issue here: https://github.com/casperstorm/ajour/issues/51

**Comment by balthild** [comment]
It's a bug. Thanks for reporting. I'll fix it soon.

**Comment by balthild** [comment]
Should be fixed in `0.4.2`.
